dynamic_programming/fibonacci_sequence_recursion_with_memoization.py

```diff
@@ -0,0 +1,49 @@
+# Fibonacci Sequence Using Recursion and Memoization
+
+# Memoization Concept: https://en.wikipedia.org/wiki/Memoization
+# Fibonacci Concept : https://pt.wikipedia.org/wiki/Sequ%C3%AAncia_de_Fibonacci
+
+
+def fibonacci(n: int, cache: dict = dict()) -> int:
+    """
+    Return number of index n in sequence fibonacci
+    >>> [fibonacci(i) for i in range(12)]
+    [0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89]
+    >>> fibonacci(12)
+    144
+    """
+    aux = n
```

Comment:
I don't think there's any need to assign another variable, we can directly use `n`.

Reply:
Corrected, thanks for the remarks